hook.py

```diff
@@ -19,72 +22,177 @@
+    payloads = []
+
+    command, new_payloads = _use_default_inputs(entries, test, platform, test['executor']['command'])
+    payloads.extend(new_payloads)
     command = _handle_multiline_commands(command)
-    cleanup = _use_default_inputs(test, test['executor'].get('cleanup_command', ''))
-    cleanup = _handle_multiline_commands(cleanup)
 
-    encoded_command = BaseWorld.encode_string(command)
-    encoded_cleanup = BaseWorld.encode_string(cleanup)
+    cleanup, new_payloads = _use_default_inputs(entries, test, platform, test['executor'].get('cleanup_command', ''))
+    payloads.extend(new_payloads)
+    cleanup = _handle_multiline_commands(cleanup)
 
-    return (encoded_command, encoded_cleanup)
+    # TODO handle ART local files in command
+    # eg. https://github.com/redcanaryco/atomic-red-team/blob/a956d4640f9186a7bd36d16a63f6d39433af5f1d/atomics/T1022/T1022.yaml
+    # eg. https://github.com/redcanaryco/atomic-red-team/blob/fa67d8f04188955ec2c3d1532d2212f26ef5e3e8/atomics/T1502/T1502.yaml
+    return (command, cleanup, payloads)
 
 
-async def _save_ability(data_svc, entries, test):
+async def _save_ability(technique_to_tactics, entries, test):
     ability_id = hashlib.md5(json.dumps(test).encode()).hexdigest()
     at_ingested = 0
     at_total = 0
+
+    tactics_li = technique_to_tactics.get(entries['attack_technique'], ['redcanary-unknown'])
+    if len(tactics_li) > 1:
+        tactic = 'multiple'
+    else:
+        tactic = tactics_li[0]
+
+    data = {
+        'id': ability_id,
+        'name': test['name'],
+        'description': test['description'],
+        'tactic': tactic,
+        'technique': {
+            'attack_id': entries['attack_technique'],
+            'name': entries['display_name']
+        },
+        'platforms': {}
+    }
     for p in test['supported_platforms']:
         at_total += 1
         if test['executor']['name'] == 'manual':
             # this test is expected to be run manually by a human, no automation is provided
             continue
-        encoded_command, encoded_cleanup = await _prepare_executor(test)
-        await data_svc.store(
-            Ability(ability_id=ability_id, tactic='redcanary', technique_id=entries['attack_technique'],
-                    technique=entries['display_name'], name=test['name'], description=test['description'],
-                    platform=PLATFORMS.get(p, 'unknown'), executor=EXECUTORS.get(test['executor']['name'], 'unknown'),
-                    test=encoded_command, cleanup=encoded_cleanup, requirements=[], parsers=[], variations=[])
-        )
+
+        command, cleanup, payloads = await _prepare_executor(entries, test, p)
+
+        executor = EXECUTORS.get(test['executor']['name'], 'unknown')
+        platform = PLATFORMS.get(p, 'unknown')
+        ex_dict = {'command': command}
+        ex_dict['payloads'] = payloads
+        ex_dict['cleanup'] = cleanup
+        data['platforms'][platform] = {executor: ex_dict}
```

Comment:
You can probably make this into a single definition:
```Python
ex_dict = dict(command=command, payloads=payloads if payloads else None, ...)
```

Reply:
fixed in 6ea28af